dojo/tools/bugcrowd_api/api_client.py

```diff
@@ -0,0 +1,134 @@
+            "filter[program]": program,
+            "page[limit]": 100,
+            "page[offset]": 0,
+            "include": "monetary_rewards,target",
+            "sort": "submitted-desc",
+        }
+
+        if target:
+            params = params_default
+            params["filter[target]"] = target
+            params_encoded = urlencode(params)
+        else:
+            params_encoded = urlencode(params_default)
+
+        next = "{}/submissions?{}".format(self.bugcrowd_api_url, params_encoded)
+        while next != "":
+            response = self.session.get(url=next)
+            response.raise_for_status()
+            if response.ok:
+                data = response.json()
+                if len(data["data"]) != 0:
+                    yield data["data"]
+
+                # When we hit the end of the submissions, break out
+                if len(data["data"]) == 0:
+                    next = ""
+                    break
+
+                # Otherwise, keep updating next link
+                next = "{}{}".format(self.bugcrowd_api_url, data["links"]["next"])
+            else:
+                next = "over"
+
+    def test_connection(self):
+        # Request programs
+        response_programs = self.session.get(
+            url="{}/programs".format(self.bugcrowd_api_url)
+        )
+        response_programs.raise_for_status()
+
+        # Request submissions to validate the org token
+        response_subs = self.session.get(
+            url="{}/submissions".format(self.bugcrowd_api_url)
+        )
+        response_subs.raise_for_status()
+
+        if response_programs.ok and response_subs.ok:
+            data = response_programs.json().get("data")
+            data_subs = response_subs.json().get("meta")
+            total_subs = str(data_subs["total_hits"])
+
+            progs = list(filter(lambda prog: prog["type"] == "program", data))
+            program_names = ", ".join(
+                list(map(lambda p: p["attributes"]["code"], progs))
+            )
+            # Request targets to validate the org token
+            response_targets = self.session.get(
+                url="{}/targets".format(self.bugcrowd_api_url)
+            )
+            response_targets.raise_for_status()
+            if response_targets.ok:
+                data_targets = response_targets.json().get("data")
+                targets = list(
+                    filter(lambda prog: prog["type"] == "target", data_targets)
+                )
+                target_names = ", ".join(
+                    list(map(lambda p: p["attributes"]["name"], targets))
+                )
+                return f'With {total_subs} submissions, you have access to the "{ program_names }" programs, \
+                    you can use these as Service key 1 for filtering submissions \
+                        You also have targets "{ target_names }" that can be used in Service key 2'
+            else:
+                raise Exception(
```

Comment:
When[ I was talking ](https://github.com/DefectDojo/django-DefectDojo/pull/6621#discussion_r969682196)about

> It is better to give some useful information to the user about what is wrong (error code/message). 
> Like here https://github.com/DefectDojo/django-DefectDojo/blob/dc20ee1bf9f99953b50c610bd9840053e8fd12f1/dojo/tools/sonarqube_api/api_client.py#L69-L71

I meant adding information from the HTTP response. The user does not need to know only what does happen (_... test not successful ..._) but also why it does happen (_unable to do DNS resolution_ (my DNS service is down?), _unable to open TCP connection_ (I need to add firewall rule?), _HTTP 401_ (my API key expired?), _HTTP 403/404_ (mistake in URL?), ...).

Reply:
Done ! I have added the response.text to the exception message, even though I'm not able to trigger that bug to validate things, but it should provide more than enough info (also the raise for status will give some info earlier anyways, 401 will trigger that)